### PR TITLE
A couple of minor improvements to FindCairo

### PR DESCRIPTION
**Pull request by ptosco**
After committing these changes several times by mistake, it's probably time to submit them intentionally.
The current `FindCairo` module, contrary to the majority of `cmake` modules, does not allow to explicitly set the path to `cairo` headers and libraries.
Additionally, building on CentOS 7 against `conda` `cairo` libraries the `Xau` dependency is needed, but it is not automatically linked. This may well be a peculiarity of my build environment, but I think it should not hurt - let's see if test pass.

### CODE REVIEW

File: Code/cmake/Modules/FindCairo.cmake [hunk 1]
```diff
@@ -42,17 +42,28 @@ PKG_CHECK_MODULES(PC_CAIRO cairo) # FIXME: After we require CMake 2.8.2 we can p
...
 FIND_LIBRARY(CAIRO_LIBRARIES
     NAMES cairo
-    HINTS ${PC_CAIRO_LIBDIR}
+    HINTS ${CAIRO_LIBRARY_DIR}
+          ${PC_CAIRO_LIBDIR}
           ${PC_CAIRO_LIBRARY_DIRS}
 )
 
+IF (UNIX AND NOT APPLE)
+	PKG_CHECK_MODULES(PC_XAU xau) # FIXME: After we require CMake 2.8.2 we can pass QUIET to this call.
```
Comment (greglandrum) [inline Code/cmake/Modules/FindCairo.cmake:59]: We already require cmake 3.5, so we should either add `QUIET` or remove this comment

**Review by greglandrum — CHANGES_REQUESTED** [review]
one minor suggestion

File: Code/cmake/Modules/FindCairo.cmake (hunk 1 above)
Comment (ptosco) [reply Code/cmake/Modules/FindCairo.cmake]: @greglandrum done.

File: Code/cmake/Modules/FindCairo.cmake [hunk 2]
```diff
@@ -38,21 +38,32 @@
...
     NAMES cairo
-    HINTS ${PC_CAIRO_LIBDIR}
+    HINTS ${CAIRO_LIBRARY_DIR}
+          ${PC_CAIRO_LIBDIR}
           ${PC_CAIRO_LIBRARY_DIRS}
 )
 
+IF (UNIX AND NOT APPLE)
+	PKG_CHECK_MODULES(PC_XAU xau QUIET)
+	FIND_LIBRARY(XAU_LIBRARIES
```
Comment (bp-kelley) [inline Code/cmake/Modules/FindCairo.cmake:60]: Doesn't this reinforce an X dependency even if using cairo-nox?

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: Code/cmake/Modules/FindCairo.cmake (hunk 2 above)
Comment (ptosco) [reply Code/cmake/Modules/FindCairo.cmake:60]: Hmmm, good point. I'll look into linking `Xau` only if linking with `cairo` fails due to missing `Xau`.

**Comment by ptosco** [comment]
@bp-kelley @greglandrum I remove the `libXau` bit from `Code/cmake/Modules/FindCairo.cmake` as it was not actually needed.
In fact, I managed to fix the problem at my end by setting `LDFLAGS="-L/usr/lib64 -Wl,-rpath-link,$CONDA_PREFIX/lib:/usr/lib64"` before invoking `cmake`. The `conda`  C++ compiler `x86_64-conda_cos6-linux-gnu-g++` was simply failing to look into `/usr/lib64` for the system `libXau.so.6`.
I think it is still useful to enable `CAIRO_INCLUDE_DIR` and `CAIRO_LIBRARY_DIR` which are standard in `cmake` `FindXXX` modules, along with minor modernization tweaks, so I think it would be good to merge this.

**Review by greglandrum — APPROVED** [review]

**Pull request merged**